sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala

```diff
@@ -138,6 +138,10 @@ class ParquetFileFormat
       conf.setBoolean(ParquetOutputFormat.ENABLE_JOB_SUMMARY, false)
     }
 
+    require(!conf.getBoolean(ParquetOutputFormat.ENABLE_JOB_SUMMARY, false)
```

Comment:
We need to issue an `AnalysisException` here.

Reply:
`AnalysisException`? Shouldn't this be `SparkException`? By the time this runs, Spark has already analyzed, optimized, and planned the job. Doesn't seem like failing analysis is appropriate.

Reply:
`SparkException` is better. Normally, we want to issue a Spark-specific exception type.

Reply:
`SparkException` makes it sound like it's a problem that Spark caused in some way. While this is caused by user input being incorrect, in which case the suggested `IllegalArgumentException` (which `require` throws) is better imo.

Reply:
In Spark SQL, we do issue the `AnalysisException` in many similar cases. I am also fine to use `SparkException`. 

In this specific case, the users are able to control the conf to make it works. Thus, we also need to improve the message to let users know how to resolve it by changing the conf.

Reply:
There's another option which is "log @ warn and continue". If someone has changed the committer, they get the consequences. That could also permit someone with a modified committer to generate schema summaries if they chose/permitted.

IT'd simplify this patch, need the tests tweaked...I'd change the SQLConf text with the committer key to say "if the committer isn't a ParquetOutputCommitter then don't expect summaries"

Reply:
I think I'd prefer the warn & continue option. It does little good to fail so late in a job, when the caller has already indicated that they want to use a different committer. Let them write the data out since this isn't a correctness issue, and they can add a summary file later if they want. Basically, there's less annoyance and interruption by not writing a summary file than by failing a job and forcing the user to re-run near the end.

Reply:
+1 for warn and continue.

Reply:
If we issuing a warning log, we will see such a warning message for each write operation. Does it look annoying?

Reply:
yes. there is that. Options: do something complicated with a static field to only print ones. Log at debug so people only see the message if they are trying to track things down.

Reply:
I think once per write operation is fine. It's not like it is once per file.